lib/FusionInventory/Agent/Task/Inventory/Win32/Storages/HP.pm

```diff
@@ -0,0 +1,55 @@
+package FusionInventory::Agent::Task::Inventory::Win32::Storages::HP;
+
+use strict;
+use warnings;
+
+use parent 'FusionInventory::Agent::Task::Inventory::Module';
+
+use English qw(-no_match_vars);
+use UNIVERSAL::require;
+
+use FusionInventory::Agent::Tools::Storages::HP;
+
+sub _getHpacuacliFromWinRegistry {
+
+    my $Registry;
+    Win32::TieRegistry->require();
+    Win32::TieRegistry->import(
+        Delimiter   => '/',
+        ArrayValues => 0,
+        TiedRef     => \$Registry,
+    );
+
+    my $machKey = $Registry->Open('LMachine', {
+        Access => Win32::TieRegistry::KEY_READ(),
+    }) or die "Can't open HKEY_LOCAL_MACHINE key: $EXTENDED_OS_ERROR";
```

Comment:
I hate that win32 too specific code ! It is really dirty. I understand you're just re-using it from _getHpacuacliFromWinRegistry(). But this code is definitively not portable and modules should be as portable as possible. I know few other places in the agent where I should myself remove such kind of code.

Btw, can you try to use getRegistryKey() and/or getRegistryValue() API from Tools::Win32. I'll do myself if you can't manage too.

Reply:
Please do it yourself as this Registry stuff is completely foreign to me.